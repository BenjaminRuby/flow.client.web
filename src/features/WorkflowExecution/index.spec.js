import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { WorkflowExecutionContainer } from "./index";

jest.mock("./Main", () => "Main");

const match = { params: {} };
const mockfn = jest.fn();
const tasks = {
  isFetching: false,
  status: "",
  error: "",
  data: []
};

const tasksActions = {
  fetch: () => new Promise(() => {}),
  reset: mockfn
};

const workflowActions = {
  fetch: () => new Promise(() => {}),
  reset: mockfn
};

const workflowExecutionActions = {
  fetch: () => new Promise(() => {}),
  reset: mockfn
};

const workflowRevisionActions = {
  fetch: () => new Promise(() => {}),
  reset: mockfn
};

const workflow = {
  isFetching: false,
  isUpdating: false,
  isCreating: false,
  fetchingStatus: "",
  updatingStatus: "",
  creatingStatus: "",
  error: "",
  data: {}
};

const workflowExecution = {
  isFetching: false,
  status: "",
  error: "",
  data: {}
};

// TO DO: To be removed if not used in future development
// const workflowExecutionActiveNode = {
//   activeNode: {}
// };

const workflowRevision = {
  isFetching: false,
  isCreating: false,
  fetchingStatus: "",
  creatingStatus: "",
  error: "",
  dag: undefined,
  version: "",
  config: {}
};

const app = {
  activeNode: {
    dag: {
      id: "3f58d302-2a3c-4765-b606-7573d313a2fa",
      offsetX: -51.15583333333335,
      offsetY: 2.996013020833331,
      zoom: 100.01666666666667,
      gridSize: 0,
      links: [],
      nodes: [
        {
          id: "cbdc30f0-f8d5-4674-87d3-117787fe8263",
          type: "startend",
          selected: false,
          x: 300,
          y: 400,
          extras: {},
          ports: [
            {
              id: "6f1f5e66-9bc1-4675-a654-5f580036a173",
              type: "startend",
              selected: false,
              name: "right",
              parentNode: "cbdc30f0-f8d5-4674-87d3-117787fe8263",
              links: [],
              position: "right",
              nodePortId: "6f1f5e66-9bc1-4675-a654-5f580036a173"
            }
          ],
          passedName: "Start",
          nodeId: "cbdc30f0-f8d5-4674-87d3-117787fe8263"
        }
      ]
    }
  }
};

describe("WorkflowExecutionContainer --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowExecutionContainer", () => {
    const renderedValue = renderer.create(
      <WorkflowExecutionContainer
        app={app}
        match={match}
        tasks={tasks}
        tasksActions={tasksActions}
        workflow={workflow}
        workflowActions={workflowActions}
        workflowExecution={workflowExecution}
        workflowExecutionActions={workflowExecutionActions}
        workflowRevision={workflowRevision}
        workflowRevisionActions={workflowRevisionActions}
      />
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WorkflowExecutionContainer --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WorkflowExecutionContainer
        app={app}
        match={match}
        tasks={tasks}
        tasksActions={tasksActions}
        workflow={workflow}
        workflowActions={workflowActions}
        workflowExecution={workflowExecution}
        workflowExecutionActions={workflowExecutionActions}
        workflowRevision={workflowRevision}
        workflowRevisionActions={workflowRevisionActions}
      />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
