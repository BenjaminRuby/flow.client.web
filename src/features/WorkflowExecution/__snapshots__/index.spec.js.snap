// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowExecutionContainer --- Snapshot Capturing Snapshot of WorkflowExecutionContainer 1`] = `
<body>
  <div>
    <div
      class="c-workflow-execution"
    >
      <nav
        style="margin-bottom: 1rem; grid-area: header;"
      >
        <button>
          <a
            class="b-navigate-back"
            href="/activity"
          >
            <svg
              aria-hidden="true"
              class="b-navigate-back__chevron"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              style="will-change: transform;"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 16L20 6l1.4 1.4-8.6 8.6 8.6 8.6L20 26z"
              />
            </svg>
            <div
              class="b-navigate-back__text"
            >
              Back to Activity
            </div>
          </a>
        </button>
      </nav>
      <div
        class="c-time-progress-bar"
      >
        <div
          class="b-time-progress-bar"
        >
          <div
            class="b-time-progress-bar__start"
          >
            Start
          </div>
          <div
            class="b-time-progress-bar__finish"
          >
            Finish
          </div>
        </div>
      </div>
      <div
        class="c-workflow-diagram-execution"
      />
      <aside
        style="grid-area: sidebar;"
      >
        <aside
          class="c-workflow-summary"
        >
          <h1
            class="s-workflow-summary-title"
          >
            Workflow
          </h1>
          <section
            class="c-workflow-summary-details"
          >
            <dl
              class="b-workflow-summary-field"
            >
              <dt
                class="b-workflow-summary-field__key"
              >
                Summary
              </dt>
              <dd
                class="b-workflow-summary-field__value"
              >
                ---
              </dd>
            </dl>
            <dl
              class="b-workflow-summary-field"
            >
              <dt
                class="b-workflow-summary-field__key"
              >
                Description
              </dt>
              <dd
                class="b-workflow-summary-field__value"
              >
                ---
              </dd>
            </dl>
            <dl
              class="b-workflow-summary-field"
            >
              <dt
                class="b-workflow-summary-field__key"
              >
                Version
              </dt>
              <dd
                class="b-workflow-summary-field__value"
              >
                1
              </dd>
            </dl>
            <dl
              class="b-workflow-summary-field"
            >
              <dt
                class="b-workflow-summary-field__key"
              >
                Status
              </dt>
              <dd
                class="b-workflow-summary-field__value"
              >
                Not Started
                <img
                  alt="Status undefined"
                  class="b-activity-card__status-icon"
                  src="pending.svg"
                />
              </dd>
            </dl>
            <dl
              class="b-workflow-summary-field"
            >
              <dt
                class="b-workflow-summary-field__key"
              >
                Start Time
              </dt>
              <dd
                class="b-workflow-summary-field__value"
              >
                January 1st 2019, 12:00:00 am
              </dd>
            </dl>
            <dl
              class="b-workflow-summary-field"
            >
              <dt
                class="b-workflow-summary-field__key"
              >
                Duration
              </dt>
              <dd
                class="b-workflow-summary-field__value"
              >
                ---
              </dd>
            </dl>
          </section>
        </aside>
      </aside>
    </div>
  </div>
</body>
`;
